spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/ModuleCoordinates.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.core;
+
+
+import org.springframework.util.Assert;
+
+/**
+ * The {@code ModuleCoordinates} class contains <a href="https://maven.apache.org/pom.html#Maven_Coordinates">
+ * Maven coordinates</a> for a jar file containing a module.
+ * <p>
+ * To create a new instance, either use {@link Builder} to set the individual fields:
+ * <pre>
+ * new ModuleCoordinates.Builder()
+ *     .setGroupId("org.springframework")
+ *     .setArtifactId("spring-core")
+ *     .setVersion("5.0.0")
+ *     .build()
+ * </pre>
+ * ...or use {@link #parse(String)} to parse the coordinates as a colin delimited string:
```

Comment:
I'm not crazy about the Builder pattern here since all fields are required. This introduces the possibility of a runtime exception, e.g.,  ModuleCoordinates.Builder.setGroupId("foo").build();   and less code to just use the constructor.